api/client/trust.go

```diff
@@ -324,19 +345,38 @@ func (cli *DockerCli) trustedPull(repoInfo *registry.RepositoryInfo, ref registr
 		if err != nil {
 			return notaryError(repoInfo.FullName(), err)
 		}
+		repoHasReleasesRole, err := hasReleasesRole(notaryRepo)
+		if err != nil {
+			return notaryError(repoInfo.FullName(), err)
+		}
```

Comment:
Unfortunately it needs to be called after `ListTargets` and `GetTargetByName`, since both of those update the repository with the latest trust info (including the delegation info).  If we moved this test outside the `if` statement, then we'd be deciding whether it has the role based on old information which might be outdated.

We are considering changing notary to require explicit calls to `Update` first rather than implicitly calling it in the `ListTargets`, `GetTargetByName`, and `Publish` functions - if so, then `Update` and the check for the releases role can be moved outside the `if` statement.